### DISCUSSION TITLE
[Help] useEffect cann't use useReadContracts?

### DISCUSSION THREAD

**Discussion by qq378488249** [WalletConnect/walletconnect-monorepo#5209] [Developer Support]
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

![微信截图_20240724153754](https://github.com/user-attachments/assets/21e17d56-b7ec-4e3c-adb9-a351cf66429b)

**Comment by JoeyKhd** [comment]
You can't use a hook inside a hook.

Unsure of what you are trying to accomplish but you could use the following inside the useEffect:

https://wagmi.sh/core/api/actions/readContract#readcontract

**Comment by flymauntains** [comment]
Can't use a hook inside a hook. If problem is continuous, contact me. I will solved.
